src/module_wrap.cc

```diff
@@ -954,15 +955,17 @@ Maybe<URL> ResolveExportsTarget(Environment* env,
     Local<Object> target_obj = target.As<Object>();
     bool matched = false;
     Local<Value> conditionalTarget;
-    if (env->options()->experimental_conditional_exports &&
-        target_obj->HasOwnProperty(context, env->node_string()).FromJust()) {
-      matched = true;
-      conditionalTarget =
-          target_obj->Get(context, env->node_string()).ToLocalChecked();
-      Maybe<URL> resolved = ResolveExportsTarget(env, pjson_url,
-            conditionalTarget, subpath, pkg_subpath, base, false);
-      if (!resolved.IsNothing()) {
-        return resolved;
+    if (env->options()->experimental_conditional_exports) {
+      ProcessEmitWarning(env, "The conditional exports is experimental.");
+      if (target_obj->HasOwnProperty(context, env->node_string()).FromJust()) {
+        matched = true;
+        conditionalTarget =
+            target_obj->Get(context, env->node_string()).ToLocalChecked();
+        Maybe<URL> resolved = ResolveExportsTarget(env, pjson_url,
+              conditionalTarget, subpath, pkg_subpath, base, false);
+        if (!resolved.IsNothing()) {
+          return resolved;
+        }
```

Comment:
Nice work getting this in the C++! Thought that was the major tricky one here. Note the standard warning message structure, it would be goo dot ensure we are following this. Note there is also a path for this in https://github.com/nodejs/node/blob/master/lib/internal/modules/cjs/loader.js#L577.